spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/JdbcPollingChannelAdapter.java

```diff
@@ -82,14 +83,38 @@ public JdbcPollingChannelAdapter(DataSource dataSource, String selectQuery) {
 		this(new JdbcTemplate(dataSource), selectQuery);
 	}
 
+	/**
+	 * Constructor taking {@link DataSource} from which the DB Connection can be
+	 * obtained and the select query to execute to retrieve new rows.
+	 * @param dataSource Must not be null
+	 * @param selectQuery Supplier for the selectQuery query to execute
+	 * @since 5.3.1
```

Comment:
`5.2.1` then, but I'm a bit reluctant to do all this `Supplier`-based stuff.
Why then don't allow to configure a `DataSource` the same way?

What would be a problem to have just a simple setter for this `selectQuery` instead?
So, you still be able to change a query at runtime, but that is going to be in more cleaner way: really like a Control Bus command...